src/styles.ts

```diff
@@ -171,3 +176,46 @@ export interface IWhiteSpaceProps {
 export const whiteSpace = (props: IWhiteSpaceProps) => ({
   whiteSpace: props.whiteSpace,
 });
+
+/**
+ * SPACE
+ * we overwrite styled-system space function to have better space prop priority p > px > pr
+ */
+
+// @ts-ignore missing type
+export const space = ss.mapProps((props: ss.SpaceProps) => ({
+  ...props,
+  // need to pass theme to rule
+  theme: useTheme(),
```

Comment:
This looks very tricky.
Any other way we could avoid using a hook like this?

Reply:
@P0lip I could call it in the box component then pass to the style prop, which is a little less confusing

Reply:
Yes, please. 
If someone adds an 'early' return, an exception will be thrown.

Reply:
fixed